packages/evolution-common/src/services/questionnaire/types/NavigationTypes.ts

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2025, Polytechnique Montreal and contributors
+ *
+ * This file is licensed under the MIT License.
+ * License text available at https://opensource.org/licenses/MIT
+ */
+
+/**
+ * Represents a section path that includes iteration information
+ * Format: sectionShortname[(/iterationContextParts)*]
+ * Examples: 'home', 'visitedPlaces/person/123', 'segments/person/123/journey/456'
+ */
+export type NavigationSection = {
+    /**
+     * The section shortname, e.g. 'home', 'visitedPlaces', 'segments'
+     */
+    sectionShortname: string;
+    /**
+     * Optional iteration context, to be appended to the section shortname to
+     * get the full path with the current iteration context. It may include many
+     * elements to create the URL.
+     */
+    iterationContext?: string[];
+};
+
+export const sectionToUrlPath = (section: NavigationSection): string => {
+    if (!section.iterationContext || section.iterationContext.length === 0) {
+        return section.sectionShortname;
+    }
+    return `${section.sectionShortname}/${section.iterationContext.join('/')}`;
+};
+
+/**
+ * Navigation state that keeps track of the current path and iteration context
+ */
+export type NavigationState = {
+    /**
+     * The current section path including iteration context
+     */
+    currentSection: NavigationSection;
+
+    /**
+     * Stack of sections previously visited, to enable back navigation
+     */
+    navigationHistory: NavigationSection[];
+};
+
+/**
+ * Represents the result of navigating to the next section. It includes the new
+ * navigation state and optional values by path that needs to be updated by this
+ * navigation (like the active survey object update)
+ */
+export type TargetSectionResult = {
+    /**
+     * The new navigation state after the navigation
+     */
+    navigationState: NavigationState;
+    /**
+     * Optional values that need to be updated by this navigation
+     */
+    valuesByPath?: { [path: string]: unknown };
```

Comment:
Et si on en profitait, au moins pour la navigation, pour renommer ce champ à 'valuesToUpdate' ou alors juste `answersToUpdate` et l'appelant s'occuperait de mapper ça à `response` pour éviter de permettre de mettre à jour n'importe quoi dans l'entrevue (même si le backend ne le permettra pas, ça limite les besoin de connaître les détails de `response` au service de navigation)?

Aussi j'ajouterais un élément `sectionsToUpdate` pour pouvoir tracker les sections qui auront été visitées, mais sautées durant la navigation.

Reply:
oui answers to update, mais il y en a partout